kv/src/main/java/com/test/kv/lib/storage/Memory.java

```diff
@@ -0,0 +1,197 @@
+package com.test.kv.lib.storage;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.sun.org.apache.bcel.internal.generic.INSTANCEOF;
+import com.sun.tools.jdi.ConcreteMethodImpl;
+import com.test.kv.exceptions.KVKeyDoesNotExistsException;
+import com.test.kv.interfaces.Storage;
+
+/**
+ * Implementation of Storage -- In memory store with LRU
+ */
+public class Memory<KeyType, ValueType> implements Storage<KeyType, ValueType> {
+    /**
+     * storage is the actual store where we keep all key->value pairs
+     */
+    protected Map<KeyType, ValueType> storage;
+
+    /**
+     * maxKeySize is the max number of keys storage can store
+     */
+    protected int maxKeySize = 1024;
+
+    /**
+     * expireAtStorage stores the key->TTL pair for cleanup expired keys usage
+     */
+    protected Map<KeyType, Long> expireAtStorage;
```

Comment:
I think the data structure that group cache using is simpler.
https://github.com/golang/groupcache/tree/master/lru
since it's LRU already, I have a feeling that TTL is bit redundant.